### ISSUE TITLE
Could not find a version that satisfies the requirement MontagePy

### ISSUE THREAD

**Issue by david-guyon** [Caltech-IPAC/Montage#33]
Hello,

I'm trying to use the new python version of Montage (MontagePy, version 6.0). I'm working on a Debian 3.16, with both Python 2.7.9 and Python 3.4.2 and pip 18.1. When I execute ``pip install MontagePy``, I get the following error:

```
Could not find a version that satisfies the requirement MontagePy (from versions: )
No matching distribution found for MontagePy
```

In the documentation it seems that nothing else than the `pip install` command is required. Is it true? Additional details on the installation process of MontagePy would be appreciated.

Kind regards.

**Comment by olebole** [comment]
FYI, the next Debian 10 "Buster" release will come with a proper [python3-montagepy](https://packages.debian.org/buster/python3-montagepy) package.

**Comment by vaidehi-varma** [comment]
I am having the very exact issue. I am also working with Python 2 and so have the similar error. Kindly please someone give a solution for Python 2 and not for Python 3. 
Thank you!

**Comment by olebole** [comment]
Keep in mind that Python 2 will die by the end of this year. This includes a large number of important science packages. https://python3statement.org/

**Comment by vaidehi-varma** [comment]
I understand it. I am trying to use another package written for python2 and uses Montage. This is the reason why I am insisting on python2. 
For the rest of my research, I use Python 3.

**Comment by gigjozsa** [comment]
I just encountered the same strange thing. ``pip install MontagePy`` used to work fine before.

**Comment by gigjozsa** [comment]
This is under Python 3.8.5, pip 20.0.2

**Comment by gigjozsa** [comment]
But only on my laptop, so please ignore, sorry.

**Comment by justyncw** [comment]
Hello,

I am now having this same error message when trying to install MontagePy, either locally or in a mybinder notebook environment. 

`ERROR: Could not find a version that satisfies the requirement MontagePy (from versions: none)`
`ERROR: No matching distribution found for MontagePy`

pip version 23.1.2, Python version 3.8

Any advice here would be much appreciated.

**Comment by mberchon** [comment]
Hello,

I am facing the same issue

Env:
- MacOS,
- pip 23.1.2 (python 3.11)

```
ERROR: Could not find a version that satisfies the requirement MontagePy (from versions: none)
ERROR: No matching distribution found for MontagePy
```

**Comment by JohnGood** [comment]
Since MontagePy (a lot of) compiled code as shared-object libraries, an install package must be separately build for each platform and version of Python.  For Python, we use Anaconda releases for simplicity.  As of this date (June 9), Anaconda does not have a full release for Python 3.11, so the latest MontagePy available is for Python 3.10.  If there is some serious need for a 3.11 version before the next Anaconda release (which tend to be one and occasionally two a year), we can take a look at doing that using the "raw" python release.

**Comment by Oreilles** [comment]
Using macOS 14.4 and python 3.10, it doesn't work either:

```
$ python3 -V                                                                                          14:38:13
Python 3.10.13
$ pip3 install MontagePy                                                                              14:38:15
ERROR: Could not find a version that satisfies the requirement MontagePy (from versions: none)
ERROR: No matching distribution found for MontagePy
```

Installing directly from wheel also fails:

```
$ poetry add http://montage.ipac.caltech.edu/download/MontagePy-1.0.1-cp36-cp36m-macosx_10_7_x86_64.whl

Updating dependencies
Resolving dependencies... (0.3s)

Package operations: 1 install, 0 updates, 0 removals

  • Installing montagepy (1.0.1 http://montage.ipac.caltech.edu/download/MontagePy-1.0.1-cp36-cp36m-macosx_10_7_x86_64.whl): Failed

  RuntimeError

  Package http://montage.ipac.caltech.edu/download/MontagePy-1.0.1-cp36-cp36m-macosx_10_7_x86_64.whl cannot be installed in the current environment {'implementation_name': 'cpython', 'implementation_version': '3.10.13', 'os_name': 'posix', 'platform_machine': 'arm64', 'platform_release': '23.4.0', 'platform_system': 'Darwin', 'platform_version': 'Darwin Kernel Version 23.4.0: Wed Feb 21 21:45:49 PST 2024; root:xnu-10063.101.15~2/RELEASE_ARM64_T6020', 'python_full_version': '3.10.13', 'platform_python_implementation': 'CPython', 'python_version': '3.10', 'sys_platform': 'darwin', 'version_info': [3, 10, 13, 'final', 0], 'interpreter_name': 'cp', 'interpreter_version': '3_10'}

  at /opt/homebrew/Caskroom/miniconda/base/envs/default/lib/python3.10/site-packages/poetry/installation/executor.py:769 in _download_link
      765│             # Since we previously downloaded an archive, we now should have
      766│             # something cached that we can use here. The only case in which
      767│             # archive is None is if the original archive is not valid for the
      768│             # current environment.
    → 769│             raise RuntimeError(
      770│                 f"Package {link.url} cannot be installed in the current environment"
      771│                 f" {self._env.marker_env}"
      772│             )
      773│

Cannot install MontagePy.
```
